pkgs/by-name/hy/hyprland/package.nix

```diff
@@ -45,30 +46,52 @@
   enableNvidiaPatches ? false,
 }:
 let
-  info = builtins.fromJSON (builtins.readFile ./info.json);
+  inherit (builtins)
+    foldl'
+    fromJSON
+    readFile
+    ;
+  inherit (lib.asserts) assertMsg;
+  inherit (lib.attrsets) mapAttrsToList;
+  inherit (lib.lists)
+    concatLists
+    flatten
+    optionals
+    ;
+  inherit (lib.strings)
+    makeBinPath
+    optionalString
+    mesonBool
+    mesonEnable
+    ;
+
+  info = fromJSON (readFile ./info.json);
+
+  # possibility to add more adapters in the future, such as keepDebugInfo,
+  # which would be controlled by the `debug` flag
+  adapters = flatten [
+    stdenvAdapters.useMoldLinker
+  ];
```

Comment:
Nit: This strikes me as odd: Why do you flatten a list that only contains a function? This is a no-op.

Reply:
Leftover from a previous attempt at also adding `(optional debug keepDebugInfo)`. I will remove it.